commands/create_cluster.go

```diff
@@ -23,11 +23,11 @@ type cpuDefinition struct {
 }
 
 type memoryDefinition struct {
-	SizeGB int `yaml:"size_gb,omitempty"`
+	SizeGB float32 `yaml:"size_gb,omitempty"`
 }
 
 type storageDefinition struct {
-	SizeGB int `yaml:"size_gb,omitempty"`
+	SizeGB float32 `yaml:"size_gb,omitempty"`
```

Comment:
IMO the code is more easy when you only have to deal with one kind of types. At some point you have to type assert and cast information back and forth. Then it gets super messy because you have to use all kinds of different methods for 8, 16, 32 and 64 bits. It makes the code only more complicated IMO. Enough said. Do what you think is right.